### ISSUE TITLE
native geolocation deprecated on Android

### ISSUE THREAD

**Issue by mica16** [driftyco/ng-cordova#430]
I struggle to make the geolocation works on Android, using ngCordova and of course the org.apache.cordova.geolocation plugin. 

I came across this Jira: 
https://issues.apache.org/jira/browse/CB-5977

Is ngCordova aware of this deprecation?

What is the best way today to use geolocation on Android with ngCordova?
Indeed, it seems that, when the plugin is installed, Android always try to use the native way, that..isn't available any more. 

Should I write an ugly pseudo-code like: 
`if(window.cordova && isAndroid()) { useTheBrowserImplementation()}  elseif(window.cordova && isIos()) {useNgCordova())}`  ?

Thanks a lot,

Michael

**Comment by sean-hill** [comment]
I'm having this exact same issue, any updates to this?

**Comment by mikeon** [comment]
I also, didn't find solution for that issue.

**Comment by Kukiwon** [comment]
+1 seeing the same issues on Galaxy S3 mini running Android 4.4.4

**Comment by mikeon** [comment]
@Kukiwon Did you try to install Crosswalk? Because this is remedy. Also you can create function with setinterval, at first run it will fail but next one will execute properly.

**Comment by Kukiwon** [comment]
@mikeon yes i'm using crosswalk. I'll try using an interval for getting a location and will keep you posted.

**Comment by Kukiwon** [comment]
@mikeon your proposed solution using an interval did not work for me. Eventually we reinstalled Android on the phone and now it works. Looks like it was an issue with the device. Thanks again!

**Comment by askannan** [comment]
same issue as mica16... Is there a way out, or is there some other plugin that I can use?

**Comment by MikhailRoot** [comment]
I've tried cordova-plugin-geolocation with ngCordova but it always returned me Timeout error, meaning it can't get device's position in my even 5 minutes set timeout, so the final solution for Android 5.1 at least:
i've added permissions to `AndroidManifest.xml`
```
    <uses-permission android:name="android.permission.INTERNET" />
    <uses-permission android:name="android.permission.NETWORK_ACCESS" />
    <uses-permission android:name="android.permission.ACCESS_WIFI_STATE" />
    <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE" />
    <uses-permission android:name="android.permission.ACCESS_COARSE_LOCATION" />
    <uses-permission android:name="android.permission.ACCESS_FINE_LOCATION" />
    <uses-permission android:name="android.permission.ACCESS_LOCATION_EXTRA_COMMANDS" />
```
it wasn't working without `ACCESS_WIFI_STATE`  permission. Now it works even without a sim card, with wifi connected and location enabled.
In case someone will find it helpfull.
